pkg/reconciler/build/build.go

```diff
@@ -352,6 +370,64 @@ func (c *Reconciler) conditionForPod(pod *corev1.Pod, stepsCompleted []string) c
 	}
 }
 
+func (c *Reconciler) finalize(ctx context.Context, build *buildapi.Build) error {
+	if !slices.Contains(build.GetFinalizers(), BuildFinalizer) {
+		return nil
+	}
+
+	if build.Finished() {
+		keychain, err := c.KeychainFactory.KeychainForSecretRef(ctx, registry.SecretRef{
+			ServiceAccount: build.Spec.ServiceAccountName,
+			Namespace:      build.Namespace,
+		})
+		if err != nil {
+			return err
+		}
+
+		if err := c.RegistryClient.Delete(keychain, build.Status.LatestImage); err != nil {
+			//logger.Printf(errors.Wrapf(err, "Could not delete image %q", build.Status.LatestImage))
```

Comment:
How should we handle the error in that case? It seems like error logging is not supposed to happen at the reconciliation stage.

Reply:
You can log and it will print to the controller logs